### DISCUSSION TITLE
Installation stopped

### DISCUSSION THREAD

**Discussion by tolaris-technik** [hydrogen-music/hydrogen#1338] [Hydrogen on Linux]
Hi,
I installed the dev-tools, but "liblash-compat-dev" was not found.

Then I downloaded hydrogen-1.1.0-beta1.tar.gz and unpacked it to "/home/tolaris/Apps/hydrogen". 

Started the installation with tolaris@t10:~/Apps/hydrogen$ ./build.sh m x

The installation stopped with the message:
  * cmake init

 * cmake make

make: *** No rule to make "translations".  Conclusion.

What is wrong?

The system is Linux Mint 20.2.

tolaris

**Comment by theGreatWhiteShark** [comment]
When entering the command

```bash
$ sudo apt-get install qtbase5-dev qtbase5-dev-tools            \
	qttools5-dev qttools5-dev-tools libqt5xmlpatterns5-dev  \
	libarchive-dev libsndfile1-dev libasound2-dev liblo-dev \
	libpulse-dev libcppunit-dev liblrdf-dev                 \
	liblash-compat-dev librubberband-dev
```

in order to install all build dependencies, `apt-get` won't install anything unless all provided packages can be found. Depending on your Linux distribution same packages may be named differently. You can either don't install the optional dependency and remove it from the list above or you have to use `apt search` to lookup the proper name in your distro and replace it. LASH is an optional one but many of the others are required. Afterwards remove the `build` folder and hit `mkdir build && cd build && cmake .. && make` again.

**Reply by tolaris-technik** [reply]
I had installed step by step sucessfully:
sudo apt install qtbase5-dev qtbase5-dev-tools
sudo apt install qttools5-dev qttools5-dev-tools libqt5xmlpatterns5-dev
sudo apt install libarchive-dev libsndfile1-dev libasound2-dev liblo-dev
sudo apt install libpulse-dev libcppunit-dev liblrdf-dev

Only this one failed:
sudo apt install liblash-compat-dev librubberband-dev libjack-jackd2-dev

Here are the commands due to your documentation:
tolaris@t10:~/Apps/hydrogen$ ./build.sh
usage ./build.sh [cmds list]
cmds may be
   r[m]     => all built, temp and cache files
   c[lean]  => remove cache files
   m[ake]   => launch the build process
   mm       => launch the build process using ccache
   mt       => launch the build process with clang tidy checks enabled
   d[oc]    => build html documentation
   g[raph]  => draw a dependencies graph
   h[elp]   => show the build options
   x[exec]  => execute hydrogen
   t[ests]  => execute tests
   p[kg]    => build source package
   z        => build using ccache and run from tree
ex: ./build.sh r m pkg x
tolaris@t10:~/Apps/hydrogen$ ./build.sh m x
 * cmake init

./build.sh: Zeile 36: cmake: Befehl nicht gefunden
tolaris@t10:~/Apps/hydrogen$ cd build
tolaris@t10:~/Apps/hydrogen/build$ sudo make install
make: *** Keine Regel, um „install“ zu erstellen.  Schluss.


It is not working.

**Reply by theGreatWhiteShark** [reply]
Ah, okay. You also need the `cmake` package which is missing in the listing above.

```bash
sudo apt-get install cmake
```

**Reply by cme** [reply]
One thing that this reminds me of, is that the appveyor configuration is sort of the definitive build recipe at the moment, and it doesn't use `build.sh` so there's also always the possibility that may fail or become stale. It would be good to unify everything (or at least incorporate as much as possible into `build.sh` and use that from Appveyor)

**Reply by theGreatWhiteShark** [reply]
Speaking of the definite build recipe: It just comes to my mind that @trebmuh reported #923 where he observes that only the English manual is included by default. I never checked whether this is true for the Windows and Mac builds we offer via Github.

**Reply by tolaris-technik** [reply]
There is still something missing:
tolaris@t10:~/Apps/Hydrogen$ ./build.sh m x
 * cmake init

-- The C compiler identification is GNU 9.3.0
-- The CXX compiler identification is unknown
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
CMake Error at CMakeLists.txt:17 (PROJECT):
  No CMAKE_CXX_COMPILER could be found.

  Tell CMake where to find the compiler by setting either the environment
  variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
  to the compiler, or to the compiler name if it is in the PATH.


-- Configuring incomplete, errors occurred!
See also "/home/tolaris/Apps/Hydrogen/build/CMakeFiles/CMakeOutput.log".
See also "/home/tolaris/Apps/Hydrogen/build/CMakeFiles/CMakeError.log".
tolaris@t10:~/Apps/Hydrogen$

**Reply by theGreatWhiteShark** [reply]
```bash
sudo apt-get install build-essential
```

should do the job

**Reply by tolaris-technik** [reply]
So it is. Thanks.
